pkg/controller/applicationlayer/applicationlayer_controller.go

```diff
@@ -304,16 +304,25 @@ func (r *ReconcileApplicationLayer) Reconcile(ctx context.Context, request recon
 
 // updateApplicationLayerWithDefaults populates the applicationlayer with defaults.
 func updateApplicationLayerWithDefaults(al *operatorv1.ApplicationLayer) {
-	defaultLogIntervalSeconds := int64(5)
-	defaultLogRequestsPerInterval := int64(-1)
+	var defaultLogIntervalSeconds int64 = 5
+	var defaultLogRequestsPerInterval int64 = -1
+	var defaultLogCollectionDisabled operatorv1.LogCollectionStatusType = "Disabled"
+	var defaultWebApplicationFirewallDisabled operatorv1.WAFStatusType = "Disabled"
```

Comment:
We should use the typed constants for these rather than the string "Disabled"

Reply:
We can't pass a pointer to a typed constant hence this is in place.

Reply:
sorry, what I meant was not to use a pointer to a constant, rather to do this:

```
disabled := operatorv1.Disabled
```

and then 

```
al.Spec.LogCollection.CollectLogs = &disabled
```

Reply:
done.